index.js

```diff
@@ -183,35 +182,36 @@ module.exports = {
           })
         })
       }, 'object'),
-      accept: valid.async(function (invite, cb) {
+      accept: valid.async((invite, cb) => {
         // remove surrounding quotes, if found
-        if(isObject(invite))
-          invite = invite.invite
+        if (isObject(invite)) { invite = invite.invite }
 
-        if (invite.charAt(0) === '"' && invite.charAt(invite.length - 1) === '"')
-          invite = invite.slice(1, -1)
+        if (invite.charAt(0) === '"' && invite.charAt(invite.length - 1) === '"') { invite = invite.slice(1, -1) }
         var opts
         // connect to the address in the invite code
         // using a keypair generated from the key-seed in the invite code
-        if(ref.isInvite(invite)) { //legacy ivite
-          if(ref.isLegacyInvite(invite)) {
+        if (ref.isInvite(invite)) { // legacy invite
+          if (ref.isLegacyInvite(invite)) {
             var parts = invite.split('~')
-            opts = ref.parseAddress(parts[0])//.split(':')
-            //convert legacy code to multiserver invite code.
+            opts = ref.parseAddress(parts[0])// .split(':')
+            // convert legacy code to multiserver invite code.
             var protocol = 'net:'
-            if (opts.host.endsWith(".onion"))
-              protocol = 'onion:'
-            invite = protocol+opts.host+':'+opts.port+'~shs:'+opts.key.slice(1, -8)+':'+parts[1]
+            if (opts.host.endsWith('.onion')) { protocol = 'onion:' }
+            invite = protocol + opts.host + ':' + opts.port + '~shs:' + opts.key.slice(1, -8) + ':' + parts[1]
           }
         }
 
-        opts = ref.parseAddress(ref.parseInvite(invite).remote)
+        const parsedInvite = ref.parseInvite(invite)
+        if (!parsedInvite || !parsedInvite.remote) {
+          return cb(new Error(`ssb-invite failed to parse invite ${invite}`))
```

Comment:
This is the only functional change I added. Previously the `ref.parseInvite` was returning `null` which caused the code to explode. (This was the ipv6 based invites problem)